### PR TITLE
[SPARK-45316][CORE][SQL] Add new parameters `ignoreCorruptFiles`/`ignoreMissingFiles` to `HadoopRDD` and `NewHadoopRDD`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to add new parameters `ignoreCorruptFiles`/`ignoreMissingFiles` to `HadoopRDD` and `NewHadoopRDD`, and set it to the current value of:
- `spark.files.ignoreCorruptFiles`/`ignoreMissingFiles` in Spark `core`,
- `spark.sql.files.ignoreCorruptFiles`/`ignoreMissingFiles` when the rdds created in Spark SQL.

### Why are the changes needed?
1. To make `HadoopRDD` and `NewHadoopRDD` consistent to other RDDs like `FileScanRDD` created by Spark SQL that take into account the SQL configs `spark.sql.files.ignoreCorruptFiles`/`ignoreMissingFiles`.
2. To improve user experience with Spark SQL, so, users can control ignoring of missing files without re-creating spark context.

### Does this PR introduce _any_ user-facing change?
Yes, `HadoopRDD`/`NewHadoopRDD` invoked by SQL code such hive table scans respect the SQL configs `spark.sql.files.ignoreCorruptFiles`/`ignoreMissingFiles` and don't respect the core configs `spark.files.ignoreCorruptFiles`/`ignoreMissingFiles`.

### How was this patch tested?
By running the affected tests:
```
$ build/sbt "test:testOnly *QueryPartitionSuite"
$ build/sbt "test:testOnly *FileSuite"
$ build/sbt "test:testOnly *FileBasedDataSourceSuite"
```

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala [hunk 1]
```diff
@@ -101,13 +101,34 @@ class HadoopRDD[K, V](
     inputFormatClass: Class[_ <: InputFormat[K, V]],
     keyClass: Class[K],
     valueClass: Class[V],
-    minPartitions: Int)
+    minPartitions: Int,
+    ignoreMissingFiles: Boolean)
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala:105]: let's include the ignore corrupted file as well.

File: core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala [hunk 2]
```diff
@@ -74,9 +74,26 @@ class NewHadoopRDD[K, V](
     inputFormatClass: Class[_ <: InputFormat[K, V]],
     keyClass: Class[K],
     valueClass: Class[V],
-    @transient private val _conf: Configuration)
+    @transient private val _conf: Configuration,
+    ignoreMissingFiles: Boolean)
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala:78]: ditto

File: docs/sql-migration-guide.md [hunk 3]
```diff
@@ -25,6 +25,7 @@ license: |
 ## Upgrading from Spark SQL 3.5 to 4.0
 
 - Since Spark 4.0, the default value of `spark.sql.maxSinglePartitionBytes` is changed from `Long.MaxValue` to `128m`. To restore the previous behavior, set `spark.sql.maxSinglePartitionBytes` to `9223372036854775807`(`Long.MaxValue`).
+- Since Spark 4.0, any read of SQL tables takes into the SQL configs `spark.sql.files.ignoreCorruptFiles`/`spark.sql.files.ignoreMissingFiles` instead of the core config `spark.files.ignoreCorruptFiles`/`spark.files.ignoreMissingFiles`.
```
Comment (superdupershant) [inline docs/sql-migration-guide.md:28]: "takes into **account** the SQL"
can say consideration instead of account as well.

**Review by superdupershant — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
All GAs passed: https://github.com/MaxGekk/spark/actions/runs/6308660727/job/17138958521
Merging to master. Thank you, @cloud-fan and @superdupershant for review.

**Pull request closed without merging**